Automattic/wp-calypso
# NPS Survey: check eligibility from endpoint

This PR queries against the undocumented NPS eligibility endpoint to determine whether or not to show the survey. To prevent unnecessary requests, the query only occurs when they user's session is "lucky" (1 out of 5000 times).

To test:
* Setup your console to see debug messages from the feature: `localStorage.setItem( 'debug', 'calypso:nps-survey' )`
* Reload Calypso and watch your console... unless you're lucky, you will likely see the following debug messages:
```
calypso:nps-survey Checking NPS eligibility... +0ms
calypso:nps-survey ...Session was not lucky +18ms
```
* Modify `NPS_SURVEY_RAND_MAX` in `client/state/nps-survey/constants.js` such that it's easier to receive the survey (using a value like `1`)
* Reload Calypso once more. This time you should see one of two things. If you've already taken or dismissed the survey, you should see a message like this:
```
calypso:nps-survey Checking NPS eligibility... +0ms
calypso:nps-survey ...Eligibility returned from endpoint. +693ms Object {display_survey: false, seconds_until_eligible: 10540794}
```
* Otherwise you should see the survey prompt, and in your console see a message like this:
```
calypso:nps-survey Checking NPS eligibility... +0ms
calypso:nps-survey ...Eligibility returned from endpoint. +722ms Object {display_survey: true, seconds_until_eligible: 0}
```
